pkg/ottl/grammar.go

```diff
@@ -174,7 +176,7 @@ func (c *compareOp) String() string {
 // comparison represents an optional boolean condition.
 type comparison struct {
 	Left  value     `parser:"@@"`
-	Op    compareOp `parser:"@OpComparison"`
+	Op    compareOp `parser:"@(OpComparison | Equal Equal)"`
```

Comment:
Does this allow for `3 = = 4.9`?

Reply:
I'll test that but probably.

Reply:
It does.  Not sure what to do about that yet since we use `.Elide` to remove all whitespace tokens.

Reply:
Moving the `=` token after `OpComparison` so that `==` will always be matched first seems to work.

Reply:
Nice!